lib/spack/spack/test/install.py

```diff
@@ -357,10 +327,8 @@ def test_partial_install_keep_prefix(install_mockery, mock_fetch, monkeypatch, w
     spack.store.STORE.failure_tracker.clear(s, True)
 
     s.package.set_install_succeed()
-    s.package.stage = MockStage(s.package.stage)
     s.package.do_install(keep_prefix=True)
     assert s.package.spec.installed
-    assert not s.package.stage.test_destroyed
```

Comment:
Same here, this property can't be tested. If it's set to true, it would be in a sub-process and we would never know in the test.